### ISSUE TITLE
`require-explicit-slots` reports on slots with dynamic names

### ISSUE THREAD

**Issue by andreww2012** [vuejs/eslint-plugin-vue#2589]
**Checklist**

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have read the [FAQ](https://eslint.vuejs.org/user-guide/#faq) and my problem is not listed.

**Tell us about your environment**

- **ESLint version:** 9.13.0
- **eslint-plugin-vue version:** 9.30.3
- **Vue version:** 3.5.12
- **Node version:** 
- **Operating System:** 

**What did you do?**

Hello! `vue/require-explicit-slots` reports on slots with dynamic names. Paste the following code [on the rule page](https://eslint.vuejs.org/rules/require-explicit-slots.html):

```vue
<template>
  <div>
    <slot :name="`${hello} ${world}`"></slot>

    <slot :name="'foo'"></slot>
  </div>
</template>

<script setup lang="ts">
const hello = 'hello';
const world = 'world';

defineSlots<{
  foo: () => unknown;
}>();
</script>
```

*(Obviously, I don't expect the second slot not be reported - I just decided to test both options)*

I think that either (A) such slots should just be ignored or (B) the rule should require the presence of something like `{[K in string]: () => unknown;}` in `defineSlots` (although the intersections `&` are not recognized at this moment).

**What did you expect to happen?**

**What actually happened?**

**Repository to reproduce this issue**
